runtime/src/iree/hal/drivers/null/README.md

```diff
@@ -0,0 +1,66 @@
+# Null HAL Driver (`null`)
+
+This is a skeleton HAL driver that implements most interfaces needed by a HAL
+driver. It contains notes that can be useful to implementers, stubs for
+easy copy/paste/renaming, and some default behavior in places that many
+implementations can use (such as file queue operations). It doesn't run anything
+and nearly all methods return `UNIMPLEMENTED` errors: this can be used to
+incrementally build a driver by running until you hit an UNIMPLEMENTED,
+implementing it, and running again. Note however that the HAL is fairly regular
+and starting with the files and running down the methods is often much easier
+to do than the trial-and-error approach: if you can implement command buffer
+fill (memset) you can often implement copy (memcpy) as well at the same time.
```

Comment:
Could also add a section on testing.

* Example commands for compiling and running, for "running until you hit an UNIMPLEMENTED"
* Template CTS CMake file, like https://github.com/iree-org/iree/blob/main/runtime/src/iree/hal/drivers/metal/cts/CMakeLists.txt with all tests under `EXCLUDED_TESTS` to start

Reply:
in the future - I think we need to rework the CTS a bit - for now I'm ok with people needing to know how to use IREE to run anything in order to implement a HAL backend :)